viewing_party/party.py

```diff
@@ -1,23 +1,136 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    # if any of the values in our inputs are falsy 
+    result = None
+    
+
+    if not title or not genre or not rating:
+        return None 
+    else:
+        new_movie = {}
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie 
+
+    
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for key in user_data["watchlist"]:
+        if key["title"] == title:
+            user_data["watchlist"].remove(key)
+            user_data["watched"].append(key)
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+    
+def get_watched_avg_rating(user_data):
+    avg_rating = 0
+    total_rating = 0
+    if user_data["watched"]:
+        for movie in user_data["watched"]:
+            total_rating += movie["rating"]
+        avg_rating = total_rating/len(user_data["watched"])
+        return avg_rating
+    else:
+        return 0.0

```

Comment:
I'd suggest flipping the order of the conditions around to handle the special case (ie, checking if something is falsy) first, so that the code can quickly check and get out of there. This arrangement is referred to as a "guard clause" or a "guard condition." Notice that it lets us move the main focus of the function (calculating the average rating) out of an indented block, letting it stand out more clearly.

```suggestion
    if not user_data["watched"]:
        return 0.0
    avg_rating = 0
    total_rating = 0
    for movie in user_data["watched"]:
        total_rating += movie["rating"]
    avg_rating = total_rating/len(user_data["watched"])
    return avg_rating
```